src/new/sendTransaction.ts

```diff
@@ -494,4 +509,182 @@ class SendTransaction extends EventEmitter {
   }
 }
 
-export default SendTransaction;
+/**
+ * Check the tx data and propose inputs and outputs to complete the transaction.
+ * We will only check a single token
+ *
+ * @param {IStorage} storage
+ * @param {IDataTx} dataTx
+ * @param {IUtxoSelectionOptions} options
+ */
+export async function prepareSendTokensData(
+  storage: IStorage,
+  dataTx: IDataTx,
+  options: IUtxoSelectionOptions = {},
+): Promise<Pick<IDataTx, 'inputs'|'outputs'>> {
+  const token = options.token || HATHOR_TOKEN_CONFIG.uid;
+  const utxoSelection = options.utxoSelectionMethod || bestUtxoSelection;
+  const newtxData: Pick<IDataTx, 'inputs'|'outputs'> = { inputs: [], outputs: [] };
+  let outputAmount = 0;
+
+  // Calculate balance for the token on the transaction
+  for (const output of dataTx.outputs) {
+    if (isDataOutputCreateToken(output)) {
+      // This is a mint output
+      // Since the current transaction is creating the token we can safely ignore it
+      continue;
+    }
+    const outputToken = output.token || HATHOR_TOKEN_CONFIG.uid;
+    if (outputToken !== token) {
+      // This output is not for the token we are looking for
+      continue;
+    }
+    outputAmount += output.value;
+  }
+
+  if (options.chooseInputs) {
+    if (outputAmount === 0) {
+      // We cannot process a target amount of 0 tokens.
+      throw new Error('Invalid amount of tokens to send.');
+    }
+
+    // We will choose the inputs to fill outputAmount.funds
+    const newUtxos = await utxoSelection(storage, token, outputAmount);
```

Comment:
If `newUtxos.inputs === []`, it means we can't fulfill this request for utxos, so should we throw an error?

Reply:
Did the same check but with `newUtxos.amount < outputAmount`